app/controllers/migration_controller.rb

```diff
@@ -103,4 +104,15 @@ def story_params
                edm_hasViews_attributes: [%i(id _destroy dc_creator dc_description dc_type dcterms_created media media_cache remove_media)]
              })
   end
+
+  # Method to remove unwanted media files, this is called after validation errors
```

Comment:
The problem is that `#valid?` is being called twice on each `EDM::WebResource`.

This can be fixed by changing, in `ORE::Aggregation`:
```ruby
    validates_associated :edm_aggregatedCHO, :edm_isShownBy, :edm_hasViews
```
... to:
```ruby
    validates_associated :edm_aggregatedCHO
```